packages/core/src/types/languages.ts

```diff
@@ -78,7 +78,7 @@ export interface LanguageFile extends Disposable {
 	>(
 		rule: AnyRule<RuleAbout, OptionsSchema>,
 		options: InferredObject<OptionsSchema>,
-	): NormalizedRuleReport[];
+	): Promise<NormalizedRuleReport[]>;
```

Comment:
🙅 I really don't like this. Changing rules' main functions to be async ripples up through a lot of the main flow.

I tried this out because cspell's initialization logic is async. But that's only needed once to set up the `DocumentValidator`s. Which, by the way, should only be set up once - not per-rule.

A compromise for now: allow rules to be defined as either a rule definition object _or_ an async function that returns one. This way they can capture async setup logic in that closure.